### PR TITLE
Add docs

### PR DESCRIPTION
**Pull request by tmigot**
@abelsiqueira This is a first attempt to make a doc based on the one from [`DCISolver.jl`](https://juliasmoothoptimizers.github.io/DCISolver.jl/dev/).

I don't exactly know how the algorithm work, so feel free to fill in the blanks in the fine-tune section.

### CODE REVIEW

File: docs/src/benchmark.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+``` @example ex1
+problems = (NLSProblems.eval(problem)() for problem in filter(x -> x != :NLSProblems, names(NLSProblems)) )
+```
+
+We compare here CaNNOLeS with `trunk` (A. R. Conn, N. I. M. Gould, and Ph. L. Toint (2000). Trust-Region Methods, volume 1 of MPS/SIAM Series on Optimization.) implemented in [JSOSolvers.jl](https://github.com/JuliaSmoothOptimizers/JSOSolvers.jl) on a subset of NLSProblems problems.
+``` @example ex1
+using CaNNOLeS, JSOSolvers
+```
+ To make stopping conditions comparable, we set `Ipopt`'s parameters `dual_inf_tol=Inf`, `constr_viol_tol=Inf` and `compl_inf_tol=Inf` to disable additional stopping conditions related to those tolerances, `acceptable_iter=0` to disable the search for an acceptable point.
```
Comment (abelsiqueira) [inline docs/src/benchmark.md:23]: Right above it was `trunk`, now it's Ipopt and below is `tron` and `trunk`.

File: docs/src/benchmark.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+  :neval_grad => "number of ∇f evals", 
+  :neval_jac => "number of ∇c evals", 
+  :neval_jprod => "number of ∇c*v evals", 
+  :neval_jtprod  => "number of ∇cᵀ*v evals", 
+  :neval_hess  => "number of ∇²f evals", 
+  :elapsed_time => "elapsed time"
+)
+perf_title(col) = "Performance profile on NLSProblems w.r.t. $(string(legend[col]))"
+
+styles = [:solid,:dash,:dot,:dashdot] #[:auto, :solid, :dash, :dot, :dashdot, :dashdotdot]
```
Comment (abelsiqueira) [inline docs/src/benchmark.md:75]: The comment can be removed

**Review by abelsiqueira — CHANGES_REQUESTED** [review]
Thanks a lot, this is great work. I just noticed some issues.

File: docs/src/benchmark.md (hunk 1 above)
Comment (tmigot) [reply docs/src/benchmark.md]: Thanks! I fixed that one.

**Review by abelsiqueira — COMMENTED** [review]
Thanks!

**Review by abelsiqueira — APPROVED** [review]

**Pull request merged**